Multiple small fixes

ede-somogyi-algolia opened the pull request:
1. Clarified usage of the `newURL` object of type `URL`.
  The old version also works since `URL.prototype.toString()` returns the `href` property as well, this just makes the code less ambiguous.
2. Fixed grammatical and spelling errors
3. Renamed variables:
  - `sendedChunk` to `sentChunks`
  - `sendedRecords` to `sentRecords`
  - `failedChunk` to `failedChunks`
  - `countFailedShunks` to `countFailedChunks`
  - `calkMaxNumberOfEntries` to `calcMaxNumberOfEntries`
4. Reworded BM module text resources to make them clearer

sbellone commented:
Oh you could also add the `const updateLogType = 'LastProductSyncLog';` refactor in `roductIndexJob.js`

ede-somogyi-algolia commented:
> Oh you could also add the `const updateLogType = 'LastProductSyncLog';` refactor in `roductIndexJob.js`

Added changes from `productsIndexJob.js` and also added the `udateLogType` to `categoryIndexJob.js`.
